src/i965_encoder_common.h

```diff
@@ -0,0 +1,517 @@
+    INTEL_MEDIA_STATE_16x16_PU_MODE_DECISION             = 26,
+    INTEL_MEDIA_STATE_8x8_PU	                         = 27,
+    INTEL_MEDIA_STATE_8x8_PU_FMODE                       = 28,
+    INTEL_MEDIA_STATE_32x32_B_INTRA_CHECK                = 29,
+    INTEL_MEDIA_STATE_HEVC_B_MBENC                       = 30,
+    INTEL_MEDIA_STATE_RESET_VLINE_STRIDE                 = 31,
+    INTEL_MEDIA_STATE_HEVC_B_PAK                         = 32,
+    INTEL_MEDIA_STATE_HEVC_BRC_LCU_UPDATE                = 33,
+    INTEL_MEDIA_STATE_ME_VDENC_STREAMIN                  = 34,
+    INTEL_MEDIA_STATE_VP9_ENC_I_32x32                    = 35,
+    INTEL_MEDIA_STATE_VP9_ENC_I_16x16                    = 36,
+    INTEL_MEDIA_STATE_VP9_ENC_P                          = 37,
+    INTEL_MEDIA_STATE_VP9_ENC_TX                         = 38,
+    INTEL_MEDIA_STATE_VP9_DYS                            = 39,
+    INTEL_MEDIA_STATE_PREPROC                            = 51,
+    INTEL_MEDIA_STATE_ENC_WP                             = 52,
+    INTEL_MEDIA_STATE_HEVC_I_MBENC                       = 53,
+    INTEL_MEDIA_STATE_CSC_DS_COPY                        = 54,
+    INTEL_MEDIA_STATE_2X_4X_SCALING                      = 55,
+    INTEL_MEDIA_STATE_HEVC_LCU64_B_MBENC                 = 56,
+    INTEL_MEDIA_STATE_MB_BRC_UPDATE                      = 57,
+    INTEL_MEDIA_STATE_STATIC_FRAME_DETECTION             = 58,
+    INTEL_MEDIA_STATE_HEVC_ROI                           = 59,
+    INTEL_MEDIA_STATE_SW_SCOREBOARD_INIT                 = 60,
+    INTEL_NUM_MEDIA_STATES                               = 61
+} INTEL_MEDIA_STATE_TYPE;
+
+struct encoder_kernel_parameter
+{
+    unsigned int curbe_size;
+    unsigned int inline_data_size;
+    unsigned int sampler_size;
+};
+
+struct encoder_scoreboard_parameter
+{
+    unsigned int mask;
+    unsigned int type;
+    unsigned int enable;
+    unsigned int walkpat_flag;
+};
+
+
+/*
+ME related defines
+*/
+#define INTEL_ENC_HME_4x    0
+#define INTEL_ENC_HME_16x   1
+#define INTEL_ENC_HME_32x   2
+
+/*
+   the definition for rate control
+*/
+#define GENERIC_BRC_SEQ         0x01
+#define GENERIC_BRC_HRD         0x02
+#define GENERIC_BRC_RC          0x04
+#define GENERIC_BRC_FR          0x08
+#define GENERIC_BRC_FAILURE     (1 << 31)
+
+enum INTEL_ENC_KERNAL_MODE
+{
+    INTEL_ENC_KERNEL_QUALITY      = 0,
+    INTEL_ENC_KERNEL_NORMAL,
+    INTEL_ENC_KERNEL_PERFORMANCE
+};
+
+enum INTEL_ENC_PRESET_MODE
+{
+    INTEL_PRESET_UNKNOWN         = 0,
+    INTEL_PRESET_BEST_QUALITY    = 1,
+    INTEL_PRESET_HI_QUALITY      = 2,
+    INTEL_PRESET_OPT_QUALITY     = 3,
+    INTEL_PRESET_OK_QUALITY      = 5,
+    INTEL_PRESET_NO_SPEED        = 1,
+    INTEL_PRESET_OPT_SPEED       = 3,
+    INTEL_PRESET_RT_SPEED        = 4,
+    INTEL_PRESET_HI_SPEED        = 6,
+    INTEL_PRESET_BEST_SPEED      = 7,
+    INTEL_PRESET_LOW_LATENCY     = 0x10,
+    INTEL_PRESET_MULTIPASS       = 0x20
+};
+/*
+   the definition for encoder status
+*/
```

Comment:
Yes, this goes along with the concern about the constants in the common file too.  If we aren't doing it now, let's keep it specific until we are ready to make the change in a later patch series.